langchain4j-oracle/src/test/java/dev/langchain4j/store/embedding/oracle/CommonTestOperations.java

```diff
@@ -47,10 +49,11 @@ final class CommonTestOperations {
 
     private CommonTestOperations() {}
 
-    private static final DataSource DATA_SOURCE;
+    private static final PoolDataSource DATA_SOURCE = PoolDataSourceFactory.getPoolDataSource();
+
     static {
         try {
-            OracleDataSource oracleDataSource = new oracle.jdbc.datasource.impl.OracleDataSource();
+            DATA_SOURCE.setConnectionFactoryClassName("oracle.jdbc.pool.OracleDataSource");
```

Comment:
JavaDocs are saying we should use the impl package:

> For data source creation using Oracle JDBC driver 21c and higher versions, we strongly recommend that your code use [oracle.jdbc.datasource.impl.OracleDataSource](https://docs.oracle.com/en/database/oracle/oracle-database/23/jajdb/oracle/jdbc/datasource/impl/OracleDataSource.html), instead of this class, which has become a thin wrapper of the former class and will be gradually phased out.

https://docs.oracle.com/en/database/oracle/oracle-database/23/jajdb/oracle/jdbc/pool/OracleDataSource.html